recipes/arrow/all/conandata.yml

```diff
@@ -1,22 +1,22 @@
 sources:
   "10.0.0":
-    url: "https://github.com/apache/arrow/archive/apache-arrow-10.0.0.tar.gz"
-    sha256: "2852b21f93ee84185a9d838809c9a9c41bf6deca741bed1744e0fdba6cc19e3f"
+    url: "https://www.apache.org/dyn/closer.lua/arrow/arrow-10.0.0/apache-arrow-10.0.0.tar.gz?action=download"
```

Comment:
why not https://downloads.apache.org/arrow/arrow-10.0.0/apache-arrow-10.0.0.tar.gz?

Reply:
We can use it only for the latest release.
But old releases are removed from downloads.apache.org. For example, https://downloads.apache.org/arrow/arrow-1.0.0/apache-arrow-1.0.0.tar.gz doesn't exist.
Old releases are moved to http://archive.apache.org/dist/ .

https://www.apache.org/dyn/closer.lua can handle both cases:
* The latest releases: downloads.apache.org or mirrors
* Old releases: archive.apache.org

Reply:
So why not using https://dist.apache.org/repos/dist/release/arrow instead? This url has all releases as documented in https://infra.apache.org/release-download-pages.html#current-and-older-releases

Reply:
What about https://dlcdn.apache.org/arrow/arrow-10.0.0/ 

![image](https://user-images.githubusercontent.com/16867443/204403564-6a6d17c1-e715-4138-b2c3-c8291ec99f40.png)

Reply:
It works until Apache Arrow 11.0.0 is released.
If Apache Arrow 11.0.0 is released, the URL will be not found like https://dlcdn.apache.org/arrow/arrow-1.0.0/ .

If we use https://www.apache.org/dyn/closer.lua , we can use the same URL for both the latest release (10.0.0/10.0.1) and old releases (1.0.0/2.0.0/...).

Reply:
Why do they need to be the same?

We should absolutely prefer the official distribution so lets fix those to point to https://dlcdn.apache.org/arrow
and leave the rest (which are super old) alone.

I think thats best of both worlds

Reply:
Is it possible to add some mirror in case not available?

How different are those packages, because the sha256 is not the same.

Reply:
I agree with @kou, we should provide the most future proof link, not something we know will break in the near future for sure.

Reply:
You'll need to help those of us who dont follow this project.

That's what they claim the official URL is on their website so I am not sure why thats not enough? What other importation information do you have that says that site will be removed?

I am sorry I am not well informed about the Apache projects multiple layers of redirects but it's not at all clear why the future release would be different.

The "release policy" https://www.apache.org/legal/release-policy.html#release-types so to only use the https://downloads.apache.org/arrow/ page  and that only has 

![image](https://user-images.githubusercontent.com/16867443/204627492-81ead88a-d41b-447d-bb81-c326ad499166.png)

So the rest shouldnt even be considered official and the old way should be good enough?

Again future releases "must" go there so why dont you expect them too?

Reply:
> Why do they need to be the same?

If we don't use the same URL for the latest release and old releases, we need to update archive URL after a new upstream version (e.g. 11.0.0) is released. The arrow recipe is broken when a new upstream version is released. I think that it increases maintenance cost of the arrow recipe.

FYI: There are still some old releases (6.0.2, 7.0.0, ...) in https://downloads.apache.org/arrow/ because upstream is discussing their maintenance policy: https://lists.apache.org/thread/7mgwr02h1f7zvghym1kljyb32s50vx1o
If upstream decides that "we maintain the latest 3 major releases", https://downloads.apache.org/arrow/ will keep only 10.X.Y, 9.X.Y and 8.X.Y. Old releases, 7.X.Y and 6.Y.Z, will be removed.

> Is it possible to add some mirror in case not available?

https://downloads.apache.org/ is the original and https://dlcdn.apache.org/ is one of mirrors.
https://www.apache.org/dyn/closer.lua returns one of available (maybe faster) mirrors.

> How different are those packages, because the sha256 is not the same.

The "official" source archives were prepared: https://github.com/apache/arrow/blob/master/dev/release/02-source.sh
They are not just "git archive"-ed.

> That's what they claim the official URL is on their website so I am not sure why thats not enough?

Could you show the URL of "their website"?  https://www.apache.org/dyn/closer.lua/arrow/arrow-10.0.0/ ?
https://arrow.apache.org/install/ is "their website". If there are some pages that use https://dlcdn.apache.org/ or https://downloads.apache.org/ directly in https://arrow.apache.org/ , they should be fixed to use https://www.apache.org/dyn/closer.lua .

FYI: I'm one of Apache Arrow developers.

> What other importation information do you have that says that site will be removed?

https://downloads.apache.org/arrow/ and https://dlcdn.apache.org/arrow/ just mirrors https://dist.apache.org/repos/dist/release/arrow/ . If upstream removes https://dist.apache.org/repos/dist/release/arrow/arrow-10.0.0/ , https://downloads.apache.org/arrow/arrow-10.0.0/ and https://dlcdn.apache.org/arrow/arrow-10.0.0/ are also removed.

Upstream released 10.0.1 (FYI: I was the release manager of upstream's 10.0.0 and 10.0.1) recently. So upstream should remove https://dist.apache.org/repos/dist/release/arrow/arrow-10.0.0/ by running https://github.com/apache/arrow/blob/master/dev/release/post-07-remove-old-artifacts.sh :

https://infra.apache.org/release-distribution.html#archival

> Each project's distribution directory should contain the latest release in each branch that is currently under development. When development ceases on a version branch, the PMC should remove links to releases of that branch from their download directory.

If I run the script, the current arrow recipe is broken.

> So the rest shouldnt even be considered official and the old way should be good enough?

"The result" means https://archive.apache.org/dist/arrow/arrow-1.0.0/ , https://archive.apache.org/dist/arrow/arrow-2.0.0/ and so on, right?
If so, they are "official".

"the old way" means https://github.com/apache/arrow/archive/apache-arrow-1.0.0.tar.gz , https://github.com/apache/arrow/archive/apache-arrow-2.0.0.tar.gz and so on, right?
If so, it's not preferred because these archives aren't "official". (not "Releases" https://www.apache.org/legal/release-policy.html#release-types )

> Again future releases "must" go there so why dont you expect them too?

Please see my answer of "Why do they need to be the same?".

I hope that I could explain missing information...

Reply:
Thank you so much for all these details. They make it much easier to get it a decision.
Releasing the official sources is top notch important which is why I asked for all this.

I'll circle back after we review this :)